transfer

jirhiker opened the pull request:
<!--
- Title the PR with the JIRA project & ticket number and short description (e.g., BDMS-123: Example display bug), or NO TICKET
- Keep sections short and scannable.
-->
### **Why**
_This PR addresses the following problem / context:_
- Use bullet points here

### **How**
_Implementation summary - the following was changed / added / removed:_
- Use bullet points here

### **Notes**
_Any special considerations, workarounds, or follow-up work to note?_
-  ⚠️ 🚨 🦺  For safety, I changed the database port the testing suite uses.  port=54321. Updated docker-compose.yml

jirhiker committed:
feat: add permissions transfer to the transfer process

jacob-a-brown committed:
fix: remvove duplicate 'Sangre de Cristo Center' entry in lexicon

jacob-a-brown committed:
fix: remove duplicate 'Lamy MDWCA' entry from lexicon

jacob-a-brown committed:
fix: remove duplicate organizations from lexicon

jacob-a-brown committed:
refactor: put all mapped organizations into one file

keeping the organization mapping in a single file is easier for maintenance
and to standardize names in Ocotillo since it'll enable quick comparisons

jacob-a-brown committed:
fix: remove duplicate organizations from lexicon

jirhiker committed:
feat: refactor permission transfer logic and improve chunk processing

jacob-a-brown committed:
refactor: use single table to map companies to organizations

this uses the new single table rather than two tables that contain
the same information

jirhiker committed:
feat: refactor permission creation logic for water level and chemistry samples

jacob-a-brown reviewed (approved):
Besides codex' comments about `chunk_by_size` I think that it looks good. I need to head out for the evening, and we are trying to get the transfer initiated tonight, so I'm approving it so it can be merged and transferred after codex' comments are addressed (whether changes need to be made or not)

jirhiker committed:
refactor: improve error handling and logging in transfer processes

jirhiker committed:
feat: enhance value mapping and error handling in water levels transfer process

jirhiker committed:
refactor: improve organization mapping logging in contact transfer process

jirhiker committed:
feat: optimize groundwater level processing and improve error handling in water levels transfer

jacob-a-brown committed:
refactor: standardize Santa Fe County names in lexicon

use SFC for all santa fe county organizations for standardized nomenclature

jacob-a-brown committed:
note: write note about company --> organization mapping

jacob-a-brown committed:
fix: validate organization enum in contact transfer

this was originally used and I removed it in the refactor. I added it
back to ensure validation errors are caught if an organization is not
in the lexicon

jacob-a-brown committed:
refactor: rename 'Other' note type to 'General'

this is the nomenclature used by AMP and in the feature files, so
this change improves consistency across the codebase and teams

jirhiker committed:
feat: improve error handling and streamline database interactions in water levels transfer

jacob-a-brown committed:
refactor: move well_construction_notes to polymorphic notes table

This field used to be standalone, but with the creation and implementation
of the polymorphic notes table it has been moved there

jirhiker committed:
feat: add stratigraphy transfer and improve value mapping error handling

jacob-a-brown committed:
fix: use "General" for general notes in thing model

It used to be called "Other" but has been changed to "General"

jacob-a-brown committed:
refactor: return specific note types rather than all notes at once

The feature file requests specific note types. By returning all notes
at once, duplicate entries will be returned. Furthermore, some note
types only apply to specific thing types. So, general notes are
returned for all thing types, but then for each thing type specific
notes will be returned

jacob-a-brown committed:
refactor: update well notes test for general notes

jacob-a-brown committed:
fix: update and standardize categories

Use consistent naming conventions for organizations in lexicon and data mapper.

jirhiker committed:
feat: enhance error handling and refactor well purposes extraction in well transfer

jirhiker committed:
feat: update lexicon value retrieval to include row context in well transfer

jirhiker committed:
feat: refactor well transfer to improve formation zone handling and error logging

jirhiker committed:
feat: enhance error handling in transfer processes and refactor error capture methods

jirhiker committed:
Merge branch 'transfer' into jab-contact-organization-refactor

jirhiker committed:
Formatting changes

jirhiker committed:
Merge pull request #269 from DataIntegrationGroup/jab-contact-organization-refactor

NO TICKET: refactor mapped companies --> organizations

jacob-a-brown committed:
fix: remove well construction notes from CreateWell

these notes are no longer a field in the thing table and need
to be handled separately. they can be added back later on and handled
by the polymorphic notes table, but for
the time being they impede the transfer process

jirhiker committed:
feat: add metrics for permissions and stratigraphy transfers in transfer process

jacob-a-brown committed:
refactor: transfer well and location notes to polymorphic notes table

Add relevant notes as enumerated in well-notes.feature and remove
well_construction_notes from the well transfer as they are now
stored in the polymorphic notes table with note_type "Construction"

jacob-a-brown committed:
feat: add Coordinate as note type to lexicon

jirhiker committed:
feat: add metrics for permissions and stratigraphy transfers in transfer process

jirhiker committed:
feat: add organization validation and improve data retrieval in transfer processes

jirhiker committed:
feat: restore main function call in transfer script

jirhiker committed:
feat: update aquifer retrieval to indicate creation status

jirhiker committed:
feat: enhance formation code handling in well transfer process

jirhiker committed:
feat: skip tests relying on external services and handle None aquifer name

jirhiker committed:
feat: optimize well transfer process by removing redundant checks and enhancing logging

jirhiker committed:
feat: update lexicon value mapping in transfer utilities to remove default None handling

jirhiker committed:
feat: refactor aquifer system handling in well transfer process and enhance error logging

jirhiker committed:
feat: reduce timeout for TIGER data retrieval and enable verbose logging in pre-commit config

jirhiker committed:
feat: enhance organization validation and improve logging verbosity in transfer process

jirhiker committed:
feat: enhance organization validation and improve logging verbosity in transfer process

jirhiker committed:
feat: remove debug print statement from utility function in transfer process

jirhiker committed:
Stream after-hook processing in WellTransferer

jirhiker committed:
Formatting changes

jirhiker committed:
Merge pull request #273 from DataIntegrationGroup/codex/improve-execution-speed-of-welltransferer

Stream after-hook processing in WellTransferer

jirhiker committed:
feat: optimize data processing in transfer functions and enhance logging

jirhiker committed:
feat: add validation error handling and improve error logging in transfer process

jirhiker committed:
feat: refactor measuring point height handling and improve validation in transfer process

jirhiker committed:
Merge branch 'transfer' into bdms-233-notes-transfer-and-updates

jirhiker committed:
Merge pull request #272 from DataIntegrationGroup/bdms-233-notes-transfer-and-updates

BDMS 233: notes transfer and updates

jirhiker committed:
feat: improve data validation and error handling in transfer process

jirhiker committed:
feat: enhance asset transfer process with improved point ID handling and session management

jirhiker committed:
feat: improve error handling and logging in transfer processes

jirhiker committed:
feat: update database port configuration for safety and improve dotenv loading

jirhiker committed:
feat: update PostgreSQL port configuration for development environment

jirhiker committed:
feat: update test configuration for PostgreSQL connection and service settings

jirhiker committed:
feat: reorganize test configuration for improved readability and structure

jirhiker committed:
feat: update test configuration for PostGIS service and environment variables

jirhiker committed:
feat: enhance contact transfer functionality with point ID filtering and add test cases

jirhiker committed:
feat: update test configuration to ignore transfer tests and rename test files for clarity

jirhiker committed:
feat: add temporary backwards compatibility for well construction notes and refactor measuring point height validation

jirhiker committed:
feat: rename step method to asset_step for clarity and improve error handling in photo retrieval
